services/py-api/src/service/hackathon_service.py

```diff
@@ -70,3 +84,22 @@ async def check_capacity_register_admin_participant_case(self) -> bool:
 
         # Check against the hackathon capacity
         return number_ant_teams < self._team_repo.MAX_NUMBER_OF_VERIFIED_TEAMS_IN_HACKATHON
+
+    async def check_capacity_register_random_participant_case(self) -> bool:
+        """Calculate if there is enough capacity to register a new random participant. Capacity is measured in max
+        number of verified teams in the hackathon. This is the Capacity Check 1 from the Excalidraw 'Adding a
+        participant workflow'"""
+
+        # Fetch number of verified random participants
+        verified_random_participants = await self._participant_repo.get_verified_random_participants_count()
+
+        # Fetch number of verified registered teams
+        verified_registered_teams = await self._team_repo.get_verified_registered_teams_count()
+
+        # Calculate the anticipated number of teams if a new random participant is added
+        number_ant_teams = verified_registered_teams + ceil(
```

Comment:
Correct me if I'm mistaken but the logic here is incorrect. `ceil` rounds up meaning that you're accounting for an additional team which is not fully populated.

Reply:
Never mind. Logic is good.